server/apps/postgres-app/src/main/java/org/apache/james/PostgresJmapModule.java

```diff
@@ -0,0 +1,79 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james;
+
+import org.apache.james.jmap.api.change.EmailChangeRepository;
+import org.apache.james.jmap.api.change.Limit;
+import org.apache.james.jmap.api.change.MailboxChangeRepository;
+import org.apache.james.jmap.api.upload.UploadUsageRepository;
+import org.apache.james.jmap.memory.change.MemoryEmailChangeRepository;
+import org.apache.james.jmap.memory.change.MemoryMailboxChangeRepository;
+import org.apache.james.jmap.memory.upload.InMemoryUploadUsageRepository;
+import org.apache.james.mailbox.AttachmentManager;
+import org.apache.james.mailbox.MessageIdManager;
+import org.apache.james.mailbox.RightManager;
+import org.apache.james.mailbox.inmemory.InMemoryMailboxSessionMapperFactory;
+import org.apache.james.mailbox.store.StoreAttachmentManager;
+import org.apache.james.mailbox.store.StoreMessageIdManager;
+import org.apache.james.mailbox.store.StoreRightManager;
+import org.apache.james.mailbox.store.mail.AttachmentMapperFactory;
+import org.apache.james.vacation.api.NotificationRegistry;
+import org.apache.james.vacation.api.VacationRepository;
+import org.apache.james.vacation.api.VacationService;
+import org.apache.james.vacation.memory.MemoryNotificationRegistry;
+import org.apache.james.vacation.memory.MemoryVacationRepository;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Scopes;
+import com.google.inject.name.Names;
+
+public class PostgresJmapModule extends AbstractModule {
```

Comment:
Should not be placed in postgres-app module.

in `server/container/guice/postgres-common` maybe?

Reply:
Should I put these bindings in PostgresMailboxModule and PostgresDataModule?

Reply:
separate `PostgresJmapModule` class IMO. See `CassandraJmapModule`.

Reply:
EmailChangeRepository and MailboxChangeRepository are in CassandraMailboxModule :((

Reply:
do they belong to Jmap module or mailbox module?

Reply:
> EmailChangeRepository and MailboxChangeRepository are in CassandraMailboxModule :((

Using memory one as of now is OK.

> do they belong to Jmap module or mailbox module?

Not mailbox module IMO.

Reply:
But I am not sure TBH :)) Here is just for IT first so I would not be picky.